### PR TITLE
feat(eslint-plugin): add new rule against A.Url

### PR DESCRIPTION
**Pull request by kresimir-coko**
Pretty straightforward, this PR adds a new rule that throws an error when `new A.Url()` is added to the code.

I didn't go with an autofix solution because we don't have any preexisting mentions, and because the returned object has different methods on it compared to the vanilla URL object.

### CODE REVIEW

**Comment by bryceosterhaus** [comment]
> we don't have any preexisting mentions

Hmm, since we don't have any existing cases of this in liferay-portal I think we are probably better off just not adding this rule for now. If for some reason we see it popping up, we can add it, but it seems like people already don't use A.Url.

That seem okay?

**Comment by kresimir-coko** [comment]
> That seem okay?

Yea, sure, although [Brian himself did comment](https://github.com/brianchandotcom/liferay-portal/pull/107047#issuecomment-918144372) that it should be added so 🤷

**Comment by bryceosterhaus** [comment]
@kresimir-coko I know we don't have any in portal, but have you run it in liferay-portal yet? I just want to make sure we don't have any false positives or errors thrown

File: projects/eslint-plugin/rules/aui/lib/rules/no-url.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+ */
+
+const message = 'use the vanilla URL class instead of A.Url';
+
+module.exports = {
+	create(context) {
+		return {
+			NewExpression(node) {
+				if (
+					node.callee.object &&
```
Comment (bryceosterhaus) [inline projects/eslint-plugin/rules/aui/lib/rules/no-url.js:13]: I think we would also need to make sure `node.callee` exists. Unless it always exists in NewExpression nodes?

Also, I think we would also need to check for just `A.Url`, without the `new`. I can imagine someone might accidentally forget the `new` and so we should warn that as well.

**Comment by kresimir-coko** [comment]
Hey @bryceosterhaus I've improved the rule by addressing your notes. I've ran the rule inside `liferay-portal` and it produced no errors.

**Review by bryceosterhaus — APPROVED** [review]
LGTM!

**Pull request merged**